php/Router.php

```diff
@@ -35,6 +35,8 @@ public function __construct( $plugin ) {
 	 */
 	public function init() {
 		add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_editor_assets' ] );
+		$new_AMP_Block_Validated_URL_Post_Type = new AMP_Block_Validated_URL_Post_Type( new Plugin( __FILE__ ) );
```

Comment:
Nice instantiation of the class. 

Is it necessary to instantiate `Plugin` again? It was instantiated in:
https://github.com/Freelancer-Martin/AMP-gutenberg-dynamic-block/blob/953754ab955570fa3d6c2b75691486fc9acff36f/block-scaffolding.php#L26

Typically, variable names wouldn't be lower snakecase, like `$amp_block`, instead of `$new_AMP_Block_Validated_URL_Post_Type`.